graphql_compiler/cost_estimation/statistics.py

```diff
@@ -0,0 +1,211 @@
+        raise NotImplementedError()
+
+    def __repr__(self):
+        """Return a human-readable str representation of the CompilerEntity object."""
+        return self.__str__()
+
+    @abstractmethod
+    def get_class_count(self, class_name):
+        """Return how many objects have, or inherit, the given class name.
+
+        Args:
+            class_name: str, either a vertex class name or an edge class name defined in the
+                        GraphQL schema.
+
+        Returns:
+            - int, number of objects with, or inheriting the given class name
+
+        Raises:
+            AssertionError, if statistic for class_name does not exist.
+        """
+        raise NotImplementedError()
+
+    @abstractmethod
+    def get_edge_count_between_vertex_pair(
+        self, vertex_out_class_name, vertex_in_class_name, edge_class_name
+    ):
+        """Return number of edges of given class connecting two vertex types.
+
+        For cost estimation, sometimes we need to predict traversal costs between two vertex types
+        using an edge type that connects the vertices' abstract types. If this statistic isn't
+        provided, it can be roughly estimated using multiple calls of get_class_count().
+
+        Args:
+            vertex_out_class_name: str, vertex class name.
+            vertex_in_class_name: str, vertex class name.
+            edge_class_name: str, edge class name.
+
+        Returns:
+            - int, the count of edges if the statistic exists
+            - None otherwise
+        """
+        return None
+
+    @abstractmethod
+    def get_domain_count(self, vertex_name, field_name):
+        """Return the number of distinct values the vertex field has over all vertex instances.
+
+        This statistic helps estimate the result size of @filter directives used to restrict values
+        using equality operators like '=', '!=', and 'in_collection'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - int, domain size of the vertex field's values if the statistic exists
+            - None otherwise
+        """
+        return None
+
+    @abstractmethod
+    def get_histogram(self, vertex_name, field_name):
+        """Return a histogram for the given vertex field, providing statistics about range values.
+
+        This statistic helps estimate the result size of @filter directives used to restrict
+        numbers using inequality operators like '>', '<', and 'between'.
+
+        Preconditions:
+        1. field_name must be a valid vertex field of vertex_name
+
+        Args:
+            vertex_name: str, name of a vertex.
+            field_name: str, name of a vertex field.
+
+        Returns:
+            - list[tuple(int, int, int)], histogram as a list of tuples with 3 elements each if the
```

Comment:
Are you sure you want the inner list type not to be `tuple(float, float, int)` instead? Please see the following link for relevant info:
https://stackoverflow.com/questions/50928592/mypy-type-hint-unionfloat-int-is-there-a-number-type

Reply:
If you decide to change the docstring here please remember to change the other docstrings where this type description is mentioned.

Reply:
I'll push this is another PR shortly.